core/src/main/java/org/pentaho/di/connections/vfs/VFSConnectionProvider.java

```diff
@@ -57,4 +57,36 @@ default FileObject getDirectFile( T connectionDetails, String path ) throws Kett
     return KettleVFS.getFileObject( pvfsUrl, new Variables(), getOpts( connectionDetails ) );
   }
 
+  /**
+   * Tests if a given VFS connection is valid, optionally, with certain testing options.
+   *
+   * @param connectionDetails The VFS connection
+   * @param connectionTestOptions The testing options, or {@code null}.
+   * The default implementation exists, call other versions of method.
+   * @return {@code true} if the provided rootPath is valid; {@code false} otherwise.
+   */
```

Comment:
The statement `@return {@code true} if the provided rootPath is valid; {@code false} otherwise.` is inaccurate, imo.

The original statement where this doclet was adapted from was accurate, imo. Whether the root path being valid is included or not in the validation depends on the given options object.

```java
  /**
   * Tests if a given VFS connection is valid, optionally, with certain testing options.
   *
   * @param connectionDetails The VFS connection.
   * @param options The testing options, or {@code null}.
   * @return {@code true} if the connection is valid; {@code false}, otherwise.
   */
```

For some reason, also, the  `* @param connectionDetails The VFS connection` statement lost the ending period.

Reply:
addessed the change.